android/src/main/java/com/gantix/JailMonkey/Rooted/GreaterThan23.java

```diff
@@ -40,5 +50,23 @@ private boolean checkRootMethod2() {
         } finally {
             if (process != null) process.destroy();
         }
+    }       
+
+    private boolean checkUserIdAndExecuteSu() {
+        Process process = null;
+
+        try {
+            process = Runtime.getRuntime().exec("id");
+            BufferedReader in = new BufferedReader(new InputStreamReader(process.getInputStream()));
+            if (in.readLine().contains("uid=0")) return true;
+
+            // Android Will throw exception if permission denied when user try to invoke Su
+            process = Runtime.getRuntime().exec("su");
```

Comment:
Tested In android platform, pf will throw exception while executing SU on a non-rooted devices.